base/test/filewritermodule_tests.cpp

```diff
@@ -52,4 +52,81 @@ BOOST_AUTO_TEST_CASE(basic, * boost::unit_test::disabled())
 	delete[] pReadData;
 }
 
+BOOST_AUTO_TEST_CASE(append)
+{
+	const uint8_t* pReadData = nullptr;
+	unsigned int readDataSize = 0U;
+	ofstream myFile("./data/testOutput/fileWriterModuleSample.txt");
+	myFile << "Foo";
+	myFile.close(); 
+	auto m1 = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+	auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::ENCODED_IMAGE));
+	auto pinId = m1->addOutputPin(metadata);
+	auto m2 = boost::shared_ptr<FileWriterModule>(new FileWriterModule(FileWriterModuleProps("./data/testOutput/fileWriterModuleSample.txt", true)));
+	m1->setNext(m2);
+
+	BOOST_TEST(m1->init());
+	BOOST_TEST(m2->init());
+	const char *stringToAppend = "bar";
+	readDataSize = std::strlen(stringToAppend);
+	auto frame = m1->makeFrame(readDataSize, pinId);
+	memcpy(frame->data(), stringToAppend, readDataSize);
+
+	frame_container frames;
+	frames.insert(make_pair(pinId, frame));
+	m1->send(frames);
+	m2->step();
+	string text;
+	ifstream readFile("./data/testOutput/fileWriterModuleSample.txt");
+	std::getline(readFile, text);
+	readFile.close();
+	BOOST_TEST(text == "Foobar"); 
+
+	delete[] pReadData;
+	boost::filesystem::remove("./data/testOutput/fileWriterModuleSample.txt");
```

Comment:
see my previous comment about setup and teardown functions. 
this line should be added to both setup and teardown functions, 
imagine if this test ever failed, then the remove(..) will never be called and a "litter" will be left in development environment, so the next run the test will fail in wierd ways becuase the content of file will be foofoobar
it is very important to implement setup and teardown functions in a test to make sure that the tests always run in a prestine conext.